samza-core/src/main/java/org/apache/samza/processor/SamzaContainerController.java

```diff
@@ -107,7 +110,18 @@ public void startContainer(ContainerModel containerModel, Config config, int max
         Util.<String, MetricsReporter>javaMapAsScalaMap(metricsReporterMap),
         taskFactory);
     log.info("About to start container: " + containerModel.getContainerId());
-    containerFuture = executorService.submit(() -> container.run());
+    containerFuture = executorService.submit(() -> {
```

Comment:
Not related to this commit, but it appears that there is nothing to stop two consecutive calls to startContainer. It appears that the intention of this class is to start and stop a container once (since we do not null out container in stop and we check in shutdown if the container is created). Given that, I would suggest checking to see if startContainer has already been called here. I would probably also consolidate stopContainer and shutdown into one method - shutdown really does mean shutdown with a single use controller.

---

Also, again not strictly related to this commit: there seems to be some mismatch between startContainer and stopContainer. The former runs asynchronously while the latter runs synchronously. I would either make them consistent (and e.g. have an awaitShutdown) or distinguish the behaviors through the method name (e.g. shutdownSync).